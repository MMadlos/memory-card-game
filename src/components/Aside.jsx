import pokeballLogo from "../assets/pokeball-logo.jpg"

function Aside({ currentPoints, maxPoints, onChangeLevel, onClickInstructions }) {
	return (
		<aside>
			<div className="title-container">
				<img
					id="logo"
					src={pokeballLogo}
					alt="a normal pokeball as a logo of the page"
				/>
				<h1>Pokememory</h1>
			</div>

			<div className="level-container">
				<label htmlFor="level">Dificultad</label>
				<select
					name="level"
					id="level"
					onChange={onChangeLevel}>
					<option value="easy">Fácil</option>
					<option value="medium">Normal</option>
					<option value="hard">Difícil</option>
				</select>
			</div>

			<div className="points-container">
				<p className="points-title">Puntuación actual</p>
				<p className="points">{currentPoints}</p>
				<p className="points-title">Máxima puntuación</p>
				<p className="points">{maxPoints}</p>
			</div>

			<div className="buttons-container">
				<button
					id="instructions"
					onClick={onClickInstructions}>
					Instrucciones
				</button>
				<a
					id="github"
					href="https://github.com/MMadlos/memory-card-game"
					target="_blank"
					rel="noreferrer">
					{" "}
					<i className="fa-brands fa-github" /> Ver en github
				</a>
			</div>
		</aside>
	)
}

export default Aside
