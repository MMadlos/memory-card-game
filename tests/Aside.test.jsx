import { vi } from "vitest"
import { render, screen } from "@testing-library/react"
import { describe, it, expect } from "vitest"
import userEvent from "@testing-library/user-event"

import Aside from "../src/components/Aside"

describe("Aside component", () => {
	it("renders component", () => {
		const { container } = render(<Aside />)
		expect(container).toMatchSnapshot()
	})

	it("renders correct heading", () => {
		render(<Aside />)
		expect(screen.getByRole("heading").textContent).toMatch("Pokememory")
	})

	it("renders level selector", () => {
		render(<Aside />)

		let optionContent = []
		screen.getAllByRole("option").forEach((option) => {
			optionContent.push(option.textContent)
		})

		expect(optionContent).toStrictEqual(["Fácil", "Normal", "Difícil"])
	})

	it("should call the onClick function when clicked", async () => {
		const onClick = vi.fn()
		const user = userEvent.setup()

		render(<Aside onClickInstructions={onClick} />)

		const button = screen.getByRole("button", { name: "Instrucciones" })

		await user.click(button)

		expect(onClick).toHaveBeenCalled()
	})

	it("shouldn't call onClick function when it's not clicked", async () => {
		const onClick = vi.fn()
		render(<Aside onClickInstructions={onClick} />)
		expect(onClick).not.toHaveBeenCalled()
	})
})
