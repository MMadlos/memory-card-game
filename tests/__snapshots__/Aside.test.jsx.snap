// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Aside component > renders component 1`] = `
<div>
  <aside>
    <div
      class="title-container"
    >
      <img
        alt="a normal pokeball as a logo of the page"
        id="logo"
        src="/src/assets/pokeball-logo.jpg"
      />
      <h1>
        Pokememory
      </h1>
      <div
        class="icon-container"
      >
        <i
          class="fa-solid fa-circle-question"
        />
        <a
          href="https://github.com/MMadlos/memory-card-game"
          rel="noreferrer"
          target="_blank"
        >
          <i
            class="fa-brands fa-github"
          />
        </a>
      </div>
    </div>
    <div
      class="level-container"
    >
      <label
        for="level"
      >
        Nivel
      </label>
      <select
        id="level"
        name="level"
      >
        <option
          value="easy"
        >
          Fácil
        </option>
        <option
          value="medium"
        >
          Normal
        </option>
        <option
          value="hard"
        >
          Difícil
        </option>
      </select>
    </div>
    <div
      class="points-container"
    >
      <div
        class="current-container"
      >
        <p
          class="points-title"
        >
          Puntuación actual
        </p>
        <p
          class="points"
        >
          0
        </p>
      </div>
      <div
        class="max-container"
      >
        <p
          class="points-title"
        >
          Máxima puntuación
        </p>
        <p
          class="points"
        >
          0
        </p>
      </div>
    </div>
    <div
      class="buttons-container"
    >
      <button
        id="instructions"
      >
        Instrucciones
      </button>
      <a
        href="https://github.com/MMadlos/memory-card-game"
        id="github"
        rel="noreferrer"
        target="_blank"
      >
         
        <i
          class="fa-brands fa-github"
        />
         Ver en github
      </a>
    </div>
  </aside>
</div>
`;
